app/movie/[id].tsx

```diff
@@ -0,0 +1,740 @@
+      })
+    }
+  }, [isError, error])
+
+  useEffect(() => {
+    if (!movieId) {
+      Toast.show({
+        type: 'error',
+        text1: 'Invalid movie',
+        text2: 'Movie identifier is missing.',
+      })
+      router.back()
+    }
+  }, [movieId])
+
+  const handleGoBack = () => {
+    router.back()
+  }
+
+  const handleWatchTrailer = async () => {
+    if (!movie?.trailerUrl) {
+      return
+    }
+
+    try {
+      const canOpen = await Linking.canOpenURL(movie.trailerUrl)
+      if (canOpen) {
+        await Linking.openURL(movie.trailerUrl)
+      } else {
+        Toast.show({
+          type: 'error',
+          text1: 'Unable to open trailer',
+          text2: 'The trailer link is invalid.',
+        })
+      }
+    } catch (err) {
+      console.error('Failed to open trailer:', err)
+      Toast.show({
+        type: 'error',
+        text1: 'Unable to open trailer',
+        text2: 'Please try again later.',
+      })
+    }
+  }
+
+  if (isLoading || !movie) {
+    return (
+      <ThemedSafeAreaView
+        style={[styles.loadingContainer, { backgroundColor }]}
+      >
+        <ActivityIndicator size='large' color={accentColor} />
+      </ThemedSafeAreaView>
+    )
+  }
+
+  const subtitle = formatSubtitle(movie.genre, movie.formats)
+  const heroImage =
+    typeof movie.image === 'string' && movie.image.trim().length > 0
+      ? movie.image
+      : typeof (movie as Record<string, any>).image_url === 'string'
+        ? (movie as Record<string, any>).image_url
+        : undefined
+  const rawCast = useMemo(() => {
+    const source =
+      (movie as Record<string, any>).cast ??
+      (movie as Record<string, any>).casts ??
+      (movie as Record<string, any>).movie_cast ??
+      (movie as Record<string, any>).actors ??
+      (movie as Record<string, any>).movieActors ??
+      []
+
+    if (Array.isArray(source)) {
+      return source
+    }
+
+    if (source && typeof source === 'object') {
+      if (Array.isArray((source as Record<string, any>).items)) {
+        return (source as Record<string, any>).items
+      }
+      if (Array.isArray((source as Record<string, any>).data)) {
+        return (source as Record<string, any>).data
+      }
+      if (Array.isArray((source as Record<string, any>).actors)) {
+        return (source as Record<string, any>).actors
+      }
+    }
+
+    return []
+  }, [movie])
+
+  const castMembers = useMemo<NormalizedCastMember[]>(() => {
+    if (!Array.isArray(rawCast)) {
+      return []
+    }
+
+    const normalized: NormalizedCastMember[] = []
+
+    rawCast.forEach(member => {
+      if (!member) return
+
+      const memberObj =
+        typeof member === 'object' && member !== null ? member : {}
+      const actorObj =
+        (memberObj as Record<string, any>).actor ??
+        (memberObj as Record<string, any>).actorInfo ??
+        (memberObj as Record<string, any>).actor_data ??
+        memberObj
```

Comment:
Check what is the property that the BE returns, you don't need to check all of this